test1.py

```diff
@@ -1,88 +1,188 @@
+                    break
+                add_item = login1.add_meal_item(meal.get('id'),ingredient.get('id'),amount)
+                print(add_item)
+
+        elif meal.get('plan') == nutrition.get('id') and nutrition.get('description') in ['Thursday', 'Friday', 'Saturday', 'Sunday']:
+            kcal= 0
+            while True:
+                amount = 100
+                ingredients_req = login1.get_req('ingredient')
+                list_of_ingredients = ingredients_req.get('results')
+                ingredient = random.choice(list_of_ingredients)
+                print(ingredient)
+                add_item = login1.add_meal_item(meal.get('id'), ingredient.get('id'),amount)
+                print(add_item)
+                kcal += ingredient.get('energy') * (amount/100)
+                if kcal > 1900:
+                    print(kcal)
+                    break
+
+# ex4
+name_of_workout = "Workout38"
+add_workout = login1.add_workout_day(name_of_workout,"This is workout10")
 print(add_workout)
-# create days for the added workout
-
-get_workout = login2.get_req('workout')
-for workout in get_workout['results']:
-    if workout['name'] == name_of_workout:
-        id1 = workout['id']
-        add_day1 = login2.add_day(workout['id'],"Push",1)
-        add_day2 = login2.add_day(workout['id'],"Pull",2)
-        add_day3 = login2.add_day(workout['id'],"Legs",3)
-
-        days = login2.get_req('day')
-        list_of_days = list()
-        for day in days["results"]:
-            if day["training"] == id1:
-                list_of_days.append(day["id"])
-
-        for id in list_of_days:
-            for i in range(6):
-                add_exercise_per_day = login2.add_exercise(id,5,2)
-
-
-
-
-
 
+exercises_get = login1.get_req('exercise')
+limit = exercises_get.get('count')
+exercises_get_total = login1.get_req('exercise',0,limit)
+list_of_exercises = exercises_get_total.get('results')
+
+chest_exercises_list = list()
+back_exercises_list = list()
+legs_exercises_list = list()
+abs_exercises_list = list()
+arms_exercises_list = list()
+sholders_exercises_list = list()
+calves_exercises_list = list()
+
+for exercise in list_of_exercises:
+    if exercise.get('category') == 11:
+        chest_exercises_list.append(exercise)
+    elif exercise.get('category') == 12:
+        back_exercises_list.append(exercise)
+    elif exercise.get('category') == 9:
+        legs_exercises_list.append(exercise)
+    elif exercise.get('category') == 13:
+        sholders_exercises_list.append(exercise)
+    elif exercise.get('category') == 8:
+        arms_exercises_list.append(exercise)
+    elif exercise.get('category') == 14:
+        calves_exercises_list.append(exercise)
+    else:
+        abs_exercises_list.append(exercise)
+
+
+get_workout = login1.get_req('workout')
+for workout in get_workout.get('results'):
+    if workout.get('name') == name_of_workout:
+        id1 = workout.get('id')
+        add_day1 = login1.add_day(id1,"Push1",1)
```

Comment:
Lines 126-146 are a consisting of a lot of duplicate code. You can loop over the type of day here ('push1', 'pull1', 'legs1') here, and introduce all the code here in order to get rid of the duplicate code.